### PR TITLE
Don't clear excess entities on mapblock save (possible fix for #4067, #1425, etc.)

### PR DESCRIPTION
**Pull request by cheapie**
This *probably* shouldn't be merged as-is, at least not without some extensive testing.

Some servers have been experiencing a bug recently where the server notices there are too many entities in a mapblock, and clears them all, wiping out players too. Seeing as how nobody seems to be able to fix the offending code (it's been over a year now), I figured I'd just remove it and see what happens.

As far as I can tell, this *did* fix the issue, and even more surprisingly, didn't seem to cause any other issues. It would appear that any excess entities are just deleted the next time the mapblock is loaded anyway, and as such, it hasn't caused any issues in my singleplayer testing, while at the same time fixing these issues:
* Difficulty connecting (client dropped during connection, aka the "gray screen" bug)
* Random disconnections (client dropped while playing, bug #4067 )
* Flickering rgblightstone (entities being deleted almost immediately after creation)

If somebody comes up with a better solution, that would be much preferred to this. On the other hand, in the meantime, I would like to see some testing of this (aside from my own) to see if it's a legitimate fix or if it causes other issues.

### CODE REVIEW

File: src/environment.cpp [hunk 1]
```diff
@@ -2180,39 +2180,29 @@ void ServerEnvironment::deactivateFarObjects(bool force_delete)
...
-					u16 store_id = pending_delete ? id : 0;
-					block->m_static_objects.insert(store_id, s_obj);
+				// If static counterpart already exists in target block,
+				// remove it first.
+				// This shouldn't happen because the object is removed from
+				// the previous block before this according to
+				// obj->m_static_block, but happens rarely for some unknown
+				// reason. Unsuccessful attempts have been made to find
+				// said reason.
+				if(id && block->m_static_objects.m_active.find(id) != block->m_static_objects.m_active.end()){
```
Comment (cheapie) [inline src/environment.cpp]: While I didn't actually edit that line aside from removing an indentation level, I suppose I can fix it...
What is the space issue, and where would you prefer I split the line? I'm not terribly familiar with MT's style guidelines.

**Comment by cheapie** [comment]
I have fixed what I think is the "space issue" in question. I'm still not sure how long is "too long" (as is, it's still shorter than before the changes), nor where to split it.

**Comment by cheapie** [comment]
All style issues should, in theory, be fixed now.

**Comment by rubenwardy** [comment]
It would probably be to find out why players are being deleted here, rather than to just stop deleting too many entities

**Comment by cheapie** [comment]
Yes, that would be ideal, but nobody seems to want to do so...
(if you want to, that would be great)

Excess entities do still seem to be deleted with this, by the way, it just appears to happen when loading the mapblock rather than saving it.

**Comment by est31** [comment]
I'm against blindly merging changes that not even the author understands, nor anybody of the "reviewers". Just testing is not enough in this very complicated example I think.

**Comment by cheapie** [comment]
Me too, more or less. I've been hoping that somebody that does know what they're doing would come along and either decide if this is a good idea or come up with something that is...

**Comment by cheapie** [comment]
This has been obsoleted by #4847.

**Pull request closed without merging**